apps/frontend/src/libs/components/input/input.tsx

```diff
@@ -6,42 +6,75 @@ import {
 } from "react-hook-form";
 
 import { getValidClassNames } from "~/libs/helpers/helpers.js";
-import { useFormController } from "~/libs/hooks/hooks.js";
+import { useCallback, useFormController } from "~/libs/hooks/hooks.js";
 
 import styles from "./styles.module.css";
 
 type Properties<T extends FieldValues> = {
 	control: Control<T, null>;
-	errors: FieldErrors<T>;
+	errors?: FieldErrors<T>;
 	label: string;
 	name: FieldPath<T>;
+	options?: { label: string; value: string }[];
 	placeholder?: string;
-	type?: "email" | "password" | "text";
+	type?: "email" | "password" | "radio" | "text";
 };
 
 const Input = <T extends FieldValues>({
 	control,
 	errors,
 	label,
 	name,
+	options,
 	placeholder = "",
 	type = "text",
 }: Properties<T>): JSX.Element => {
 	const { field } = useFormController({ control, name });
 
-	const error = errors[name]?.message;
+	const error = errors && errors[name]?.message;
 	const hasError = Boolean(error);
 
+	const handleRadioChange = useCallback(
+		(event: React.ChangeEvent<HTMLInputElement>) => {
+			field.onChange(event.target.value);
+		},
+		[field],
+	);
+
 	return (
 		<label className={styles["input-wrapper"]}>
-			<span className={styles["input-label"]}>{label}</span>
+			<span
+				className={getValidClassNames(
+					styles["input-label"],
+					options && styles["radio-label"],
+				)}
+			>
+				{label}
+			</span>
 			<div className={styles["input-container"]}>
-				<input
-					className={getValidClassNames(styles["input-field"])}
-					{...field}
-					placeholder={placeholder}
-					type={type}
-				/>
+				{type === "radio" && options ? (
```

Comment:
And move it to a separate variable please
```suggestion
				{type === "radio" && options.length ? (
```